torch/csrc/jit/passes/concat_opt.cpp

```diff
@@ -590,5 +591,16 @@ bool UseVariadicCat(const std::shared_ptr<Graph>& graph) {
   return changed;
 }
 
+bool RemoveListMutationAndUseVariadicCat(const std::shared_ptr<Graph>& graph) {
```

Comment:
Looks good but as discussed offline eventually we would like to make this one pass instead of running each back and forth

Reply:
Sure. That requires refactoring some of the code to work on nodes instead of graphs on the whole. Will do that as a follow-up.